R-packages/evalcast/R/download_signals.R

```diff
@@ -0,0 +1,52 @@
+#' Download signal from covidcast
+#'
+#' This is a simple wrapper to \code{\link[covidcast]{covidcast_signal}} that is less verbose.
+#'
+#' @param ... the arguments that are passed to [covidcast::covidcast_signal()]
+#' @importFrom magrittr %>%
+#' @importFrom rlang .data
+#' @importFrom covidcast covidcast_signal
+#' @importFrom utils data
+#' @importFrom stringr str_glue str_sub
+#' @importFrom dplyr select mutate distinct left_join rename
+download_signal <- function(...) {
+  args <- list(...)
+  if (is.null(args$start_day)) {
+    msg <- stringr::str_glue(
+      "Downloading {signal} from covidcast through {end}.",
+      start = args$start_day,
+      end = args$end_day,
+      signal = args$signal,
+      .sep = " "
+    )
+  }
+  else {
+    msg <- stringr::str_glue(
+      "Downloading {signal} from covidcast for period from",
+      "{start} to {end}.",
+      start = args$start_day,
+      end = args$end_day,
+      signal = args$signal,
+      .sep = " "
+    )
+  }
+  message(msg)
+  ## Why two calls below?? Commented out one
+  ## base::suppressMessages({covidcast_signal(...)})
+  out <- base::suppressMessages({covidcast_signal(...)})
+  if (args$geo_type == "state") {
+      e  <- new.env(parent = emptyenv())
+      county_geo  <- utils::data("county_geo", package = "covidcast", envir = e)
+      state_fips <- e$county_geo %>%
+          select(.data$STATE, .data$COUNTYNAME, .data$FIPS) %>%
+          mutate(fips = str_sub(.data$FIPS, end = 2),
+                 geo_value = tolower(.data$STATE)) %>%
+          distinct(.data$geo_value, .data$fips)
+    out <- out %>%
+      left_join(state_fips, by = "geo_value") %>%
+      mutate(geo_value = .data$fips) %>%
+      select(-.data$fips)
+  }
```

Comment:
@jacobbien Here I think you're trying to convert state abbreviations to FIPS codes, if I'm understanding it correctly? 

First, we have to get rid of the dependency on `county_geo` because I just merge a PR in #122 that overwrites this and potentially results in different columns.

Second, you should just rely on the utilities in `covidcast` for mapping between geo codes and names.  I think this will do it for you:

```
out$geo_value = name_to_fips(paste0("^", abbr_to_name(toupper(out$geo_value)), "$"))
```

Here the regex anchors "^" and "$" force a match to `v` that anchors it at the start and end of the string, respectively.  (To avoid matches to things like "Virginia" matching to both "Virginia" and "West Virginia", or "Wyoming" matching to both "Wyoming" and "Wyoming County".  There might be a cleaner way to do it ... but this should work.)

Try this as an example, if you want to see the state FIPS codes: 

```
name_to_fips(paste0("^", abbr_to_name(state.abb), "$"))
```

Reply:
See (bd6912b).